src/main/scala/cognite/spark/v1/FlexibleDataModelRelationUtils.scala

```diff
@@ -514,13 +531,21 @@ object FlexibleDataModelRelationUtils {
     }
   }
 
+  sealed trait RowInstancePropertyValue
+
+  object RowInstancePropertyValue {
+    case object NotPresent extends RowInstancePropertyValue
```

Comment:
NotPresent where :thinking: when would this value be created?

Reply:
Not present in the spark row. Which is different from being present but null.

Reply:
how would that happen, it's still value for one of the keys that is defined in transformation sql? Can spark sql produce different things for null and absent, and when would it happen?
Or is it more like us for some reason reading all keys from dms, then trying to apply sparksql updates to it and nopresent is stuff we want to preserve because key isn't mentioned in sparksql?
Also in case you know, how is `select *` different in that regard from `select a,b,c` (assuming for example `a,b,c` is the exhaustive list of columns in source data)?

Reply:
See the new test case for some example sql. The properties do not have to be mentioned in the output sql.

The readme says that if a field is not mentioned, there should be no effect on that field. If it is mentioned but set to null, it should depend on ignireNullFields.

Reply:
For example:
```sql
select
'space' as space,
'extid' as externalId,
null as property
```
(present but null) vs.
```sql
select
'space' as space,
'extid' as externalId
```
(not present)

Reply:
Thanks, this clears it up a bit.
What about
```sql
select * from raw_something
```
would it behave same as
```sql
select a,b,c,d,e from raw_something
```
wrt nulls, where `a,b,c,d,e` is a guesstimate from schema inference?

Reply:
I don't really know how spark handles `select *`, but I would guess it uses the schemas specified in the `Relation` classes. In this case,
https://github.com/cognitedata/cdp-spark-datasource/blob/7dcbd1c4cdb5e4cc6441c790e704a90beebb5fcf/src/main/scala/cognite/spark/v1/FlexibleDataModelCorePropertyRelation.scala#L52

which is created by fetching the view.